arkouda/strings.py

```diff
@@ -551,3 +819,32 @@ def to_ndarray(self):
         for i, (o, l) in enumerate(zip(npoffsets, lengths)):
             res[i] = np.str_(''.join(chr(b) for b in npvalues[o:o+l]))
         return res
+
+    def save(self, prefix_path : str, mode : str='truncate') -> None:
+        """
+        Save the Strings object to HDF5. The result is a collection of HDF5 files,
+        one file per locale of the arkouda server, where each filename starts
+        with prefix_path. Each locale saves its chunk of the array to its
+        corresponding file.
+
+        Important implementation notes: (1) Strings state is saved as two datasets
+        within an hdf5 group: offsets AKA segments and values (2) save logic is
+        delegated to pdarrayIO.save_all and (3) offsets are generated server-side 
+        from the values pdarray
+
+        Parameters
+        ----------
+        prefix_path : str
+            Directory and filename prefix that all output files share
+        mode : str {'truncate' | 'append'}
+            By default, truncate (overwrite) output files, if they exist.
+            If 'append', attempt to create new dataset in existing files.
+
+        Raises
+        ------
+        ValueError 
+            Raised if the lengths of columns and values differ, or the mode is 
+            not 'truncate' or 'append'
+        """
+        arkouda.save_all(columns=[self.bytes], prefix_path=prefix_path,
```

Comment:
This is actually not needed since only the bytes array is being sent.

Reply:
You may not have gotten to this yet, but just putting it in writing that this function still needs a change. `Strings.save()` should not call `ak.save_all()`; rather, `Strings.save()` should issue the "tohdf" server command. I still think you will need to send both offsets and bytes so that Chapel can efficiently recompute the segments, but even if you can do it with only bytes, this is where the server command should be generated.

Reply:
Okay, will update to call toHdf directly. I am currently working on the alternate save strategy we discussed on Monday which requires toHdf to be called on both the values and the segments separately.

Reply:
Since you're generating new segments within toHdf already, it might be easier and cleaner just to have toHdf write out both the values and the segments in the same call.